### ISSUE TITLE
Make targets fail due to GAPPS_VARIANT not being set

### ISSUE THREAD

**Issue by fonic** [opengapps/aosp_build#37]
When OpenGApps is part of the source tree, several make targets fail, e.g. `make clean`. This is due to the fact that BoardConfig.mk or device.mk is not included when making these targets, thus GAPPS_VARIANT is undefined.

I would suggest the the following patch for `vendor/google/build/config.mk`:

```
# Device should define their GAPPS_VARIANT in device/manufacturer/product/BoardConfig.mk
ifeq ($(GAPPS_VARIANT),)
  #$(error GAPPS_VARIANT must be configured)
  # If GAPPS_VARIANT is not set, default to variant 'stock'
  GAPPS_VARIANT := stock
endif
GAPPS_VARIANT_EVAL := $(call get-gapps-variant,$(GAPPS_VARIANT))
```

Also, it should probably print a note stating that GAPPS_VARIANT was not set and now defaults to 'stock'. Since 'stock' is the recommended variant anyway, I think this something people can live with.

**Comment by Blystad** [comment]
I had this originally, when `GAPPS_VARIANT` was set in the `BoardConfig.mk` file, but that presented an issue that made me move `GAPPS_VARIANT` to `device.mk` and require it to be set (mainly that if `GAPPS_VARIANT` was set to e.g. `mini` in `BoardConfig.mk` then it wouldn't be set in `device.mk`, causing the `include` of `opengapps-packages` to provide all packages.

But I can see the issue....I don't think defaulting to stock is a good solution though. I suggest rather that we default to `none` and print a warning.

`none` doesn't exist yet, but should be fairly straightforward to add.

**Comment by fonic** [comment]
Defaulting to 'none' also sounds like a good idea.

**Comment by jamuir** [comment]
@Blystad:  do you recall the reasoning as to why the file `vendor/google/build/config.mk` is used?

That file is treated specially by the build system.  It is included by `build/core/main.mk`:

```make
# Include the google-specific config
-include vendor/google/build/config.mk
```

It also gets included by the file `vendor/google/build/opengapps-packages.mk`.

So `config.mk` is processed at least twice when a build is done.

I think the issue with `make installclean` and `make clean` could be avoided if we rename `config.mk` to something else.

**Comment by Blystad** [comment]
Originally it was to ensure that `mm` in `vendor/google/build/modules/*/` worked. I don't know if it's still needed.

**Comment by jamuir** [comment]
ok.  I will do some testing.  There may be another way to get `mm` working.

**Comment by jamuir** [comment]
I believe the original issue here was resolved by invisiblek's change (da56d8b0), but to avoid other potential issues related to automatic processing of `vendor/google/build/config.mk`, I've opened a new PR which integrates opengapps into the build tree in a different way:

https://github.com/opengapps/aosp_build/pull/129
